airbyte-integrations/connectors/source-pendo-python/source_pendo_python/streams.py

```diff
@@ -347,14 +353,50 @@ def get_json_schema(self) -> Mapping[str, Any]:
 
             full_schema = self.build_schema(full_schema, body)
             self.json_schema = full_schema
-        except requests.exceptions.RequestException:
+        except (requests.exceptions.RequestException, ValueError) as e:
+            self.logger.warning(f"Error fetching visitor metadata: {e}")
             self.json_schema = base_schema
         return self.json_schema
 
     def request_body_json(self, next_page_token: Mapping[str, Any] = None, **kwargs) -> Optional[Mapping[str, Any]]:
         source = {"visitors": {"identified": True}}
         return self.build_request_body("visitor-list", source, next_page_token)
 
+    def normalize_createdate(self, record: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        """
+        normalize the record dates to milliseconds if metadata.agent.createdate is present
+        createdate could be a timestamp in milliseconds after the epoch (UTC) or 
+        an ISO 8601 string (e.g., '2019-02-08T21:30:53.150Z')
+        """        
+        if "metadata" in record and "agent" in record["metadata"] and "createdate" in record["metadata"]["agent"]:
+            createdate = record["metadata"]["agent"]["createdate"]
+            if isinstance(createdate, str):
+                # Handle ISO 8601 strings (e.g., '2019-02-08T21:30:53.150Z')
+                try:
+                    parsed_date = pendulum.parse(createdate, strict=False).in_timezone('UTC')
+                    record["metadata"]["agent"]["createdate"] = int(parsed_date.timestamp() * 1000) # Convert to milliseconds
+                    self.logger.info(f"Converted ISO 8601 createdate to milliseconds: {record['metadata']['agent']['createdate']}")
+                except Exception as e:
+                    self.logger.error(f"Failed to parse createdate '{createdate}': {e}")
+                    raise ValueError(f"Invalid date format for createdate: {createdate}")
+            elif isinstance(createdate, int):
+                # Already in milliseconds
```

Comment:
I don't know the details of this code, but in the past I've been bitten by integers sometimes being seconds and not milliseconds. Is that a possible problem here, or are we sure from context that an integer here is always milliseconds?